templates/conv_ai/templates.yaml

```diff
@@ -0,0 +1,152 @@
+dataset: conv_ai
+templates:
+  1664cdd9-54e8-4679-821b-8013e9df197e: !Template
+    id: 1664cdd9-54e8-4679-821b-8013e9df197e
+    jinja: '{{context}}
+
+
+      {% for utterance in thread %}
+
+      - {{ utterance["userId"] }}: {{ utterance["text"] }}
+
+      {% endfor %}
+
+
+      Alice''s utterances in the previous conversation show that she was not interested
+      in the context, yes or no?
+
+      |||
+
+      {% for eval in evaluation %}
+
+      {% if "Alice" == eval["userId"] %}
+
+      {% if 3 > eval["engagement"] %}yes{% else %}no{% endif %}
+
+      {% endif %}
+
+      {% endfor %}'
+    name: engagement_alice_1
+    reference: ''
+    task_template: false
+  2d3ca9f5-60f0-459d-932f-ab30e1e79b22: !Template
+    id: 2d3ca9f5-60f0-459d-932f-ab30e1e79b22
+    jinja: '"{{context}}""
+
+
+      Would you say Bob engagement is real, given the previous context, in this conversation:
+
+      {% for utterance in thread %}
+
+      - {{ utterance["userId"] }}: {{ utterance["text"] }}
+
+      {% endfor %}
+
+      |||
+
+      {% for eval in evaluation %}
+
+      {% if "Bob" == eval["userId"] %}
+
+      {% if 3 < eval["engagement"] %}yes{% else %}no{% endif %}
+
+      {% endif %}
+
+      {% endfor %}'
+    name: engagement_bob_1
+    reference: ''
+    task_template: false
+  4390549a-3bdf-43ad-9e69-6bc380f33f01: !Template
+    id: 4390549a-3bdf-43ad-9e69-6bc380f33f01
+    jinja: '{{context}}
+
+
+      Given the previous context, who do you think is more engaged in this conversation
+      (Alice, Bob, or both):
+
+      {% for utterance in thread %}
+
+      - {{ utterance["userId"] }}: {{ utterance["text"] }}
+
+      {% endfor %}
+
+      |||
+
+      {% set alice = (evaluation|selectattr("userId", "equalto", "Alice")|first)["engagement"]
+      %}
+
+      {% set bob = (evaluation|selectattr("userId", "equalto", "Bob")|first)["engagement"]
+      %}
+
+
+      {% if alice == bob %}both{% elif alice < bob %}Bob{% else %}Alice{% endif %}'
```

Comment:
this template should be conditional: it gives a very weird prompted example where the thread has only 1 or 2 utterances (or even 0)...
<img width="646" alt="Capture d’écran 2021-06-15 à 5 12 25 PM" src="https://user-images.githubusercontent.com/16107619/122124406-40a66a00-cdfd-11eb-988d-e95c22273f25.png">
```suggestion
    jinja: '{% if thread | length >= 4%}
    {{context}}


      Given the previous context, who do you think is more engaged in this conversation
      (Alice, Bob, or both):

      {% for utterance in thread %}

      - {{ utterance["userId"] }}: {{ utterance["text"] }}

      {% endfor %}

      |||

      {% set alice = (evaluation|selectattr("userId", "equalto", "Alice")|first)["engagement"]
      %}

      {% set bob = (evaluation|selectattr("userId", "equalto", "Bob")|first)["engagement"]
      %}


      {% if alice == bob %}both{% elif alice < bob %}Bob{% else %}Alice{% endif %}
      {% endif %}'
```

Reply:
it does look weird. I should update all templates though. I'd suggest using something like ```{% if (0 < (thread | selectattr("userId", "equalto", "Bob") | list | length)) and (0 < (thread | selectattr("userId", "equalto", "Alice") | list | length)) %}```. Would that be ok?

Reply:
yeah your proposition looks good and should cover much of the cases I was worried about!
i think that would make sense to update all templates cause not sure whether it is useful who's the most engaged when someone is talking alone....

Reply:
there's still a (minor?) issue if I do this: the tests don't like the setting of alice and bob variables (```{% set alice = (evaluation|selectattr("userId", "equalto", "Alice")|first)["engagement"]```) inside of the all surrounding ```if``` and complain about "unrecognized" variables. This means I have to move the 2 ```set```at the top of the prompt, outside the ```if```. It works but will it correctly skip the incomplete examples we're trying to filter out?

Reply:
yes you'll have to do `set alice` and `set bob` outside of the if condition indeed!

The example will be filtered out as long as the template doesn't return anything (which is the case when you wrap the template inside an if condition independently of what variables you define outside the if condition)